### PR TITLE
MMU code clear up: remove core_mmu_set_info_table, core_mmu_get_user_…

### PR DESCRIPTION
**Pull request by xyzhao2018**
…pgdir from x86

Signed-off-by: xzhao3 <shirley.zhao@intel.com>

### CODE REVIEW

File: core/arch/x86_64/mm/core_mmu.c [hunk 1]
```diff
@@ -1025,7 +1025,6 @@ static unsigned long init_mem_map(struct tee_mmap_region *memory_map,
 	 */
 	vaddr_t id_map_start = (vaddr_t)__identity_map_init_start;
 	vaddr_t id_map_end = (vaddr_t)__identity_map_init_end;
-	unsigned long offs = 0;
```
Comment (jingdlu) [inline core/arch/x86_64/mm/core_mmu.c]: Can we keep this offs variable in init_mem_map?

**Review by jingdlu — COMMENTED** [review]
Don't use the same title "MMU code clear up" for this patch.

File: core/arch/x86_64/mm/core_mmu.c (hunk 1 above)
Comment (xyzhao2018) [reply core/arch/x86_64/mm/core_mmu.c]: Why keep it? Actually, it is useless and result in compile warning.

Comment (jingdlu) [reply core/arch/x86_64/mm/core_mmu.c]: Because we may need to implement ALSR for X86 as well. What kind of compile warning did you meet?

Comment (xyzhao2018) [reply core/arch/x86_64/mm/core_mmu.c]: The warning is： 
core/arch/x86_64/mm/core_mmu.c: In function ‘init_mem_map’:
core/arch/x86_64/mm/core_mmu.c:1029:16: warning: variable ‘offs’ set but not used [-Wunused-but-set-variable]
 1029 |  unsigned long offs = 0;

We can add it back when implement ALSR.

Comment (xyzhao2018) [reply core/arch/x86_64/mm/core_mmu.c]: Done, restore offs

**Review by jingdlu — APPROVED** [review]

**Pull request merged**